stable_baselines3/common/distributions.py

```diff
@@ -292,6 +292,112 @@ def log_prob(self, actions: th.Tensor) -> th.Tensor:
         return self.distribution.log_prob(actions)
 
 
+class MultiCategoricalDistribution(Distribution):
+    """
+    MultiCategorical distribution for multi discrete actions.
+
+    :param action_dims: ([int]) List of sizes of discrete action spaces.
+    """
+    def __init__(self, action_dims: List[int]):
+        super(MultiCategoricalDistribution, self).__init__()
+        self.action_dims = action_dims
+        self.distributions = None
+
+    def proba_distribution_net(self, latent_dim: int) -> nn.Module:
+        """
+        Create the layer that represents the distribution:
+        it will be the logits of the MultiCategorical distribution.
+        You can then get probabilities using a softmax.
+
+        :param latent_dim: (int) Dimension of the last layer
+            of the policy network (before the action layer)
+        :return: (nn.Linear)
+        """
+        action_logits = nn.Linear(latent_dim, sum(self.action_dims))
+        return action_logits 
+
+    def proba_distribution(self, action_logits: th.Tensor) -> 'MultiCategoricalDistribution':
+        reshaped_logits = th.split(action_logits, self.action_dims, dim=1)
+        self.distributions = [Categorical(logits=l) for l in reshaped_logits]
+        return self
+
+    def mode(self) -> th.Tensor:
+        return th.stack([th.argmax(d.probs, dim=1) for d in self.distributions])
```

Comment:
`d.mode()` does not work?

Reply:
I'm creating a vector of Categorical distributions from torch directly, not involving the CategoricalDistribution class. It seemed clearer to me this way.

Reply:
but you use d.sample for sampling...it is more a remark about consistency ;)

Reply:
But d.sample() is accessing the torch distribution no?

Reply:
Would you like me to create a helper d.mode() class in MultiCategorical and use that instead?

Reply:
oh sorry, i did not read the first comment fully, then it's fine